### PR TITLE
[BIOMAGE-2014] - Fix staging SHA

### PR DESCRIPTION
**Pull request by aerlaut**
# Description
PR #756 introduces a fix to use the appropriate commit sha to tag images. It is tagging the docker images properly, but wasn't properly picked up by the staging-candidates for the PRs. This PR fixes taht.

# Details
#### URL to issue
https://biomage.atlassian.net/browse/BIOMAGE-2014

#### Link to staging deployment URL (or set N/A)
N/A

#### Links to any PRs or resources related to this PR
https://github.com/hms-dbmi-cellenics/ui/pull/761
https://github.com/hms-dbmi-cellenics/api/pull/390
https://github.com/hms-dbmi-cellenics/worker/pull/286
https://github.com/hms-dbmi-cellenics/pipeline/pull/264

#### Integration test branch
master

# Merge checklist
Your changes will be ready for merging after all of the steps below have been completed.

### Code updates
Have best practices and ongoing refactors being observed in this PR
- [ ] Migrated any selector / reducer used to the new format.

### Manual/unit testing
- [ ] Tested changes using InfraMock locally **or** no tests required for change, e.g. Kubernetes chart updates.
- [ ] Validated that current unit tests for code work as expected and are sufficient for code coverage **or** no unit tests required for change, e.g. documentation update.
- [ ] Unit tests written **or** no unit tests required for change, e.g. documentation update.

### Integration testing
**You must check the box below** to run integration tests on the latest commit on your PR branch.
Integration tests have to pass before the PR can be merged. Without checking the box, your PR
**will not pass** the required status checks for merging.

- [ ] Started end-to-end tests on the latest commit.

### Documentation updates
- [ ] Relevant Github READMEs updated **or** no GitHub README updates required.
- [ ] Relevant Wiki pages created/updated **or** no Wiki updates required.

### Optional
- [ ] Staging environment is unstaged before merging.
- [ ] Photo of a cute animal attached to this PR.

### CODE REVIEW

**Review by cosa65 — APPROVED** [review]

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -156,13 +160,23 @@ jobs:
           # in PR synch the latest commit of the branch is found in github.event.pull_request.head.sha instead
 
           if [ "${{ github.event_name }}" = "release" ] && [ "${{ github.event.action }}" = "released" ]; then
-            echo "::set-output name=tag::${REF_ID/refs-tags-/}"
+            COMMIT_SHA=""
+            IMAGE_TAG="${REF_ID/refs-tags-/}"
           elif [ "${{ github.event_name }}" = "pull_request" ]; then
-            echo "::set-output name=tag::$REF_ID-${{ github.event.pull_request.head.sha }}"
+            COMMIT_SHA="${{ github.event.pull_request.head.sha }}"
+            IMAGE_TAG="$REF_ID-${{ github.event.pull_request.head.sha }}"
```
Comment (kafkasl) [inline .github/workflows/ci.yaml]: can you make this `IMAGE_TAG="$REF_ID-$COMMIT_SHA"`?

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -156,13 +160,23 @@ jobs:
...
           else
-            echo "::set-output name=tag::$REF_ID-$GITHUB_SHA"
+            COMMIT_SHA=$GITHUB_SHA
+            IMAGE_TAG="$REF_ID-$GITHUB_SHA"
           fi
 
+          echo "::set-output name=commit-sha::$COMMIT_SHA"
+
+          # IMAGE_TAG is used in the Build Docker Image step.
+          # We can easily build the image-tag from REF_ID and COMMIT_SHA for non-releases in the step inputs
```
Comment (kafkasl) [inline .github/workflows/ci.yaml]: all of the builds are actually releases, either to `production`, `staging-default` or just PRs. Could you specify that we cannot do that for `production` releases.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (aerlaut) [reply .github/workflows/ci.yaml]: sure

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (aerlaut) [reply .github/workflows/ci.yaml]: added

**Review by kafkasl — APPROVED** [review]

**Pull request merged**